### PR TITLE
PLAT-669 message-bus-kafka package uses new kafka with kraft in replicated fashion

### PR DESCRIPTION
**Pull request by Mark-Labuschagne**
This PR is intended to only have the changes made to the message-bus-kafka package, but will induce breaking changes if merged alone.

For that reason, the following PRs must be merged first:

- [x] https://github.com/jembi/platform/pull/220

### CODE REVIEW

File: message-bus-kafka/docker-compose.kafka-utils.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ services:
   kafdrop:
     image: obsidiandynamics/kafdrop:3.27.0
     environment:
-      KAFKA_BROKERCONNECT: 'kafka:9092'
+      KAFKA_BROKERCONNECT: 'kafka-01:9092,kafka-02:9092,kafka-03:9092'
```
Comment (bradsawadye) [inline message-bus-kafka/docker-compose.kafka-utils.yml:7]: This should be in the package-metadata.json

File: message-bus-kafka/docker-compose.kafka-utils.yml [hunk 2]
```diff
@@ -20,7 +20,7 @@ services:
     image: quay.io/cloudhut/kminion:master
     hostname: kafka-minion
     environment:
-      KAFKA_BROKERS: kafka:9092
+      KAFKA_BROKERS: kafka-01:9092,kafka-02:9092,kafka-03:9092
```
Comment (bradsawadye) [inline message-bus-kafka/docker-compose.kafka-utils.yml:23]: above comment applies

**Comment by bradsawadye** [comment]
@Mark-Labuschagne i though we agreed that we'll have three instances of kafka when running in clustered mode

File: message-bus-kafka/docker-compose.kafka-utils.yml (hunk 1 above)
Comment (Mark-Labuschagne) [reply message-bus-kafka/docker-compose.kafka-utils.yml]: Done in [2e20304](https://github.com/jembi/platform/pull/219/commits/2e2030445ea1ac3b00df4545286a138255968e41)

File: message-bus-kafka/docker-compose.kafka-utils.yml (hunk 2 above)
Comment (Mark-Labuschagne) [reply message-bus-kafka/docker-compose.kafka-utils.yml]: Done in [2e20304](https://github.com/jembi/platform/pull/219/commits/2e2030445ea1ac3b00df4545286a138255968e41)

File: message-bus-kafka/docker-compose.cluster.kafka.yml [hunk 3]
```diff
@@ -1,21 +1,21 @@
 version: '3.9'
 
 services:
-  kafka:
-    hostname: kafka-{{.Node.Hostname}}
```
Comment (nour-borgi) [inline message-bus-kafka/docker-compose.cluster.kafka.yml:5]: I think we need hostname to all the 3 instances. if we don't specify it Kafdrop will be linked to a random hostname and if kafka fails the hostname will be updated, kafdrop will not be able to reconnect to kafka after that

File: message-bus-kafka/swarm.sh [hunk 4]
```diff
@@ -19,34 +18,22 @@ function init_vars() {
...
-    )
-  fi
-
   SERVICE_NAMES=(
-    "${ZOOKEEPER_SERVICES[@]}"
     "${UTILS_SERVICES[@]}"
-    "kafka"
+    "kafka-01"
+    "kafka-02"
+    "kafka-03"
```
Comment (nour-borgi) [inline message-bus-kafka/swarm.sh:30]: I think we can put that in a variable alone KAFKA_SERVICES and we can use it later for deploy_sanity and service_reachable function as well

File: message-bus-kafka/docker-compose.dev.kafka.yml [hunk 5]
```diff
@@ -1,8 +0,0 @@
-version: '3.9'
-
-services:
-  kafka:
-    ports:
-      - target: 9092
```
Comment (nour-borgi) [inline message-bus-kafka/docker-compose.dev.kafka.yml:6]: Is this something we don't need ?

File: message-bus-kafka/swarm.sh [hunk 6]
```diff
@@ -58,35 +56,29 @@ function import_sources() {
 }
 
 function initialize_package() {
-  local kafka_dev_compose_filename=""
```
[the remark this answers is not in the document]
Comment (nour-borgi) [reply message-bus-kafka/swarm.sh:61]: I don't think we can use it, you think we should bring it back ?

File: .env.cluster [hunk 7]
```diff
@@ -58,10 +58,10 @@ JS_REPORT_DEV_MOUNT=false
 JS_REPORT_PACKAGE_PATH=
 
 # Message Bus Kafka
-KAFKA_INSTANCES=3
 # Topics should comma seperated, optional include partion and repliction values
 #   e.g. <topic>:<partions>:<replicationFactor> -> test:3:2 (defaults to <topics>:3:1)
 KAFKA_TOPICS=2xx,reprocess,3xx,metrics:3:3
+KAKFA_HOSTS=kafka-01:9092,kafka-02:9092,kafka-03:9092
```
[the remark this answers is not in the document]
Comment (nour-borgi) [reply .env.cluster:64]: ops 😮

File: message-bus-kafka/swarm.sh [hunk 8]
```diff
@@ -97,16 +89,16 @@ function initialize_package() {
     exit 1
   }
 
-  log info "Await Kafka to be running and responding"
-  config::await_service_running "kafka" "${COMPOSE_FILE_PATH}"/docker-compose.await-helper.yml "${KAFKA_INSTANCES}"
+  log info "Await Kafdrop to be running and responding"
+  config::await_service_running "kafdrop" "${COMPOSE_FILE_PATH}"/docker-compose.await-helper.yml 1
```
[the remark this answers is not in the document]
Comment (nour-borgi) [reply message-bus-kafka/swarm.sh:93]: cool done

File: message-bus-kafka/swarm.sh (hunk 6 above)
[the remark this answers is not in the document]
Comment (nour-borgi) [reply message-bus-kafka/swarm.sh:61]: done

**Review by michaelloosen — APPROVED** [review]

**Review by michaelloosen — CHANGES_REQUESTED** [review]

File: kafka-mapper-consumer/docker-compose.yml [hunk 9]
```diff
@@ -1,6 +1,7 @@
 version: '3.9'
 
 services:
+# TODO: update version once new mapper-consumer is released
   kafka-mapper-consumer:
     image: jembi/kafka-mapper-consumer
```
[the remark this answers is not in the document]
Comment (nour-borgi) [reply kafka-mapper-consumer/docker-compose.yml]: done :)

File: kafka-unbundler-consumer/docker-compose.yml [hunk 10]
```diff
@@ -1,6 +1,7 @@
 version: '3.9'
 
 services:
+# TODO: update version once new unbundler-consumer is released
   kafka-unbundler-consumer:
     image: jembi/kafka-unbundler-consumer:0.0.2
```
[the remark this answers is not in the document]
Comment (nour-borgi) [reply kafka-unbundler-consumer/docker-compose.yml]: done :)

**Review by nour-borgi — APPROVED** [review]

**Pull request merged**